lib/osml/model_runner/roles/mr_execution_role.ts

```diff
@@ -18,132 +19,102 @@ import { MRContainerConfig } from "../mr_container";
 
 /**
- * Represents the properties required to define a model runner ECS task role.
- *
+ * Defines the properties required for creating an `MRExecutionRole`.
  * @interface MRExecutionRoleProps
+ * @property {OSMLAccount} account - The OSML (OversightML) deployment account associated with this role.
+ * @property {string} roleName - The name to assign to the IAM role.
  */
 export interface MRExecutionRoleProps {
-  /**
-   * The OSML (OversightML) deployment account associated with this role.
-   *
-   * @type {OSMLAccount}
-   */
   account: OSMLAccount;
-
-  /**
-   * The name to give to the role.
-   *
-   * @type {string}
-   */
   roleName: string;
 }
 
 /**
- * Represents an MRExecutionRole construct.
+ * Constructs a new IAM role designed for ECS tasks execution within AWS,
+ * providing necessary permissions predefined for model runner operations.
+ *
+ * @class MRExecutionRole
+ * @extends {Construct}
+ * @property {IRole} role - The AWS IAM role associated with this construct.
+ * @property {string} partition - The AWS partition in which the role will operate.
  */
 export class MRExecutionRole extends Construct {
-  /**
-   * The AWS IAM role associated with this MRExecutionRole.
-   */
   public role: IRole;
-
-  /**
-   * The AWS partition to be used for this MRExecutionRole.
-   */
   public partition: string;
 
   /**
-   * The Model Runner Dataplane Configuration values to be used for this MRExecutionRole
-   */
-  public mrDataplaneConfig: MRDataplaneConfig = new MRDataplaneConfig();
-
-  /**
-   * The Model Runner Container Configuration values to be used for this MRExecutionRole
-   */
-  public mrContainerConfig: MRContainerConfig = new MRContainerConfig();
-
-  /**
-   * Creates an MRExecutionRole construct.
-   * @param {Construct} scope - The scope/stack in which to define this construct.
-   * @param {string} id - The id of this construct within the current scope.
-   * @param {MRExecutionRoleProps} props - The properties of this construct.
-   * @returns MRExecutionRole - The MRExecutionRole construct.
+   * Initializes a new instance of the `MRExecutionRole` class.
+   *
+   * @constructor
+   * @param {Construct} scope - The scope in which to define this construct, typically a CDK `Stack`.
+   * @param {string} id - A unique identifier for this construct within the scope.
+   * @param {MRExecutionRoleProps} props - The properties for configuring this role.
    */
   constructor(scope: Construct, id: string, props: MRExecutionRoleProps) {
     super(scope, id);
 
-    // Defining constants for better readability
-    const MR_CONTAINER_REPOSITORY_NAME =
-      this.mrContainerConfig.MR_CONTAINER_REPOSITORY;
-    const MR_FIRELENS_LOG_GROUP_NAME = `/aws/${this.mrDataplaneConfig.METRICS_NAMESPACE}/MRFireLens`;
-    const MR_SERVICE_LOG_GROUP_NAME = `/aws/${this.mrDataplaneConfig.METRICS_NAMESPACE}/MRService`;
+    const containerRepositoryName = new MRContainerConfig()
+      .MR_CONTAINER_REPOSITORY;
```

Comment:
Do we need this? We can remove these lines and do the `*` for resources

Reply:
Do this in the follow up PR to clean up roles